### DISCUSSION TITLE
blinking debug windows when using BeginListbox

### DISCUSSION THREAD

**Discussion by SadE54** [ocornut/imgui#4596] [New Users Build/Link/Run issues ONLY!]
Hi , 

In my application (flow based editor) I have to manage a deferred popup that is shown when the user clicks on some specific pins , and then you're able to chose some existing vars using a listbox.
I have something like this:
```
if (ImGui::BeginPopup("popup_button"))
        {
            if (pin != NULL)
            {
                switch (pin->Type)
                {
                case PinType::SysVar:
                // ...
               }
               break;

                case PinType::FlowVar:
                    ImGui::BeginListBox("##flow_var_name");
                   ...
                    ImGui::EndListBox();
                  break;
            }
       ImGui::EndPopup();
      }
```
The second case is using BeginListBox because I need to manage the content dynamically.
The popup is activated by a button linked to some pins

And it's working almost as I want , except that the in second case (using BeginListBox()) , I can see debug window opening and closing immediately at the moment the popup is opened.

 I guess ListBox() is using BeginListBox() internally, so I do not understand why it's working well in one case and noo in the other
🤔 

(I'm using ImGui 1.84 on Win 10 with DX11 backend)

**Comment by ocornut** [comment]
You are not testing the return value of BeginListBox() so during a frame where it is clipped you are closing two windows and outputting in the window at the bottom of the stack (debug window)

**Reply by SadE54** [reply]
Thanks for the fast answer, it's working well now 👌
